zilliqa/src/transaction.rs

```diff
@@ -299,6 +301,59 @@ impl SignedTransaction {
             }
         }
     }
+
+    pub fn validate_gas_limit(&self) -> Result<()> {
+        // Verify minimum gas requirement
+        match self {
+            SignedTransaction::Legacy { .. }
+            | SignedTransaction::Eip2930 { .. }
+            | SignedTransaction::Eip1559 { .. } => {
+                const MINIMUM_ETH_GAS: EvmGas = EvmGas(21_000);
+                if self.gas_limit() < MINIMUM_ETH_GAS {
+                    bail!(
+                        "Provided evm transaction gas limit is too low: {}",
+                        self.gas_limit()
+                    );
+                }
+            }
+            SignedTransaction::Zilliqa { ref tx, .. } => {
+                // Minimum gas needed for scilla run - contract invocation
+                const MINIMUM_ZIL_GAS: ScillaGas = ScillaGas(10);
+                if tx.gas_limit < MINIMUM_ZIL_GAS {
+                    bail!(
+                        "Provided scilla transaction gas limit is too low: {}",
+                        tx.gas_limit
+                    );
+                }
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+
+    pub fn validate_input_size(&self) -> Result<()> {
+        // https://eips.ethereum.org/EIPS/eip-170
+        const MAX_EVM_CONTRACT_SIZE_BYTES: usize = 24576;
+        let status = match self {
+            SignedTransaction::Legacy { tx, .. } => tx.input.len() <= MAX_EVM_CONTRACT_SIZE_BYTES,
+
+            SignedTransaction::Eip2930 { tx, .. } => tx.input.len() <= MAX_EVM_CONTRACT_SIZE_BYTES,
+
+            SignedTransaction::Eip1559 { tx, .. } => tx.input.len() <= MAX_EVM_CONTRACT_SIZE_BYTES,
+
+            SignedTransaction::Zilliqa { ref tx, .. } => {
+                // Seems there was no limit in ZQ1 (let's have one)
+                const MAX_ZILLIQA_DATA_SIZE: usize = 32 * 1024;
```

Comment:
Let's be *VERY* certain that legacy txns aren't going to hit it first. And definitely migrate to constants.rs or to the config file .. @JamesHinshelwood  for a decision.

Reply:
Calling @JamesHinshelwood  ? Do you think this is the right number?

Reply:
As mentioned in some other comment I made, I think we should remove this.